org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/model/IXtextDocument.java

```diff
@@ -21,14 +27,39 @@
  */
 public interface IXtextDocument extends IDocument, IDocumentExtension3, IReadAccess<XtextResource>, IReadAccess.Priority<XtextResource>, IWriteAccess<XtextResource> {
 
-	public <T> T getAdapter(Class<T> adapterType);
+	/**
+	 * Supported adapter types are {@link IFile}, {@link IResource}, {@link URI} and real supertypes of the implementation class, e.g. all
+	 * document extensions.
+	 */
+	default public <T> T getAdapter(Class<T> adapterType) {
+		if (adapterType.isInstance(this)) {
+			return adapterType.cast(this);
+		}
+		if (URI.class.equals(adapterType)) {
+			return adapterType.cast(getResourceURI());
+		}
+		if (IFile.class.equals(adapterType) || IResource.class.equals(adapterType)) {
+			URI resourceURI = getResourceURI();
+			if (resourceURI.isPlatformResource()) {
+				return adapterType.cast(ResourcesPlugin.getWorkspace().getRoot().getFile(new Path(resourceURI.toPlatformString(true))));
+			}
+		}
+		return Platform.getAdapterManager().getAdapter(this, adapterType);
```

Comment:
no whitelist for validator ?

Reply:
No, I don't expect a ValidationJob to be available for text buffer based documents.